clio-client/src/main/scala/org/broadinstitute/clio/client/commands/CommandDispatch.scala

```diff
@@ -2,57 +2,57 @@ package org.broadinstitute.clio.client.commands
 
 import akka.http.scaladsl.model.HttpResponse
 import com.typesafe.scalalogging.LazyLogging
-import org.broadinstitute.clio.client.ClioClientConfig
 import org.broadinstitute.clio.client.commands.Commands.{
   AddWgsUbam,
+  MoveWgsUbam,
   QueryWgsUbam
 }
 import org.broadinstitute.clio.client.parser.BaseArgs
+import org.broadinstitute.clio.client.util.IoUtil
 import org.broadinstitute.clio.client.webclient.ClioWebClient
 
-import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.concurrent.{ExecutionContext, Future}
 
-object CommandDispatch extends LazyLogging {
+class CommandDispatch(val webClient: ClioWebClient, val ioUtil: IoUtil)
+    extends LazyLogging {
 
-  def execute(command: CommandType, webClient: ClioWebClient, config: BaseArgs)(
+  private def execute(command: CommandType, config: BaseArgs)(
     implicit ec: ExecutionContext
   ): Future[HttpResponse] = {
     command match {
-      case AddWgsUbam   => AddWgsUbamCommand.execute(webClient, config)
-      case QueryWgsUbam => QueryWgsUbamCommand.execute(webClient, config)
+      case AddWgsUbam => AddWgsUbamCommand.execute(webClient, config, ioUtil)
+      case QueryWgsUbam =>
+        QueryWgsUbamCommand.execute(webClient, config, ioUtil)
+      case MoveWgsUbam => MoveWgsUbamCommand.execute(webClient, config, ioUtil)
     }
   }
 
-  def dispatch(webClient: ClioWebClient,
-               config: BaseArgs)(implicit ec: ExecutionContext): Boolean = {
+  def dispatch(
+    config: BaseArgs
+  )(implicit ec: ExecutionContext): Future[Boolean] = {
     config.command
-      .map(command => execute(command, webClient, config))
-      .exists(responseFuture => { checkResponse(responseFuture) })
+      .map(command => execute(command, config))
+      .fold(Future(false))(checkResponse)
   }
 
   def checkResponse(
     responseFuture: Future[HttpResponse]
-  )(implicit ec: ExecutionContext): Boolean = {
-    Await.result(
-      responseFuture.map[Boolean] { response =>
-        val isSuccess = response.status.isSuccess()
-
-        if (isSuccess) {
-          logger.info(
-            s"Successfully completed command." +
-              s" Response code: ${response.status}"
-          )
-          logger.info(response.toString)
-        } else {
-          logger.error(
-            s"Error executing command." +
-              s" Response code: ${response.status}"
-          )
-          logger.info(response.toString)
-        }
-        isSuccess
-      },
-      ClioClientConfig.responseTimeout
-    )
+  )(implicit ec: ExecutionContext): Future[Boolean] = {
+    responseFuture.map(response => {
```

Comment:
Nitpick: for multi-line blocks the convention is usually to pass the function in block form like:
```
responseFuture.map { response =>
  ???
}
```